src/util/redisConnector.js

```diff
@@ -0,0 +1,106 @@
+      this.password = process.env.REDIS_PASSWORD;
+      this.maxRetries = parseInt(process.env.REDIS_MAX_RETRIES || 30, 10);
+      this.timeAfterRetry = parseInt(process.env.REDIS_TIME_AFTER_RETRY_IN_MS || 10, 10);
+      this.client = new Redis({
+        host: this.host,
+        port: this.port,
+        password: this.password,
+        enableReadyCheck: true,
+        retryStrategy: (times) => {
+          if (times <= this.maxRetries) {
+            return 10 + times * this.timeAfterRetry;
+          }
+          log.error(`Redis is down at ${this.host}:${this.port}`);
+          return false; // stop retrying
+        },
+      });
+      this.client.on('ready', () => {
+        stats.increment('redis_ready', {
+          timestamp: Date.now(),
+        });
+        log.info(`Connected to redis at ${this.host}:${this.port}`);
+      });
+    }
+  },
+  /**
+   * Checks connection with redis and if not connected, tries to connect and throws error if connection request fails
+   */
+  async checkAndConnectConnection() {
+    if (!this.client) {
+      this.init();
+    }
+    else if (this.client.status !== 'ready') {
+      await Promise.race([this.client.connect(), timeoutPromise])
+    }
+  },
+  /**
+   * Used to get value from redis depending on the key and the expected value type
+   * @param {*} key key for which value needs to be extracted
+   * @param {*} isJsonExpected false if fetched value can not be json
+   * @returns value which can be json or string or number
+   *
+   */
+  async getVal(key, isJsonExpected = true) {
+    try {
+      await this.checkAndConnectConnection(); // check if redis is connected and if not, connect
+      const value = await this.client.get(key);
+      if (value) {
+        const bytes = Buffer.byteLength(value, "utf-8");
+        stats.gauge('redis_get_val_size', bytes, {
+          timestamp: Date.now()
+        });
+      }
+      return isJsonExpected ? JSON.parse(value) : value;
+    } catch (e) {
+      throw new RedisError(`Error getting value from Redis: ${e}`);
+    }
+  },
+  /**
+   * Used to set value in redis depending on the key and the value type
+   * @param {*} key key for which value needs to be stored
+   * @param {*} value value to be stored in redis
+   * @param {*} isValJson set to false if value is not a json object
+   */
+
+  async setVal(key, value, isValJson = true) {
+    const dataExpiry = 60*60; // 1 hour
+    try {
+      await this.checkAndConnectConnection(); // check if redis is connected and if not, connect
+      const valueToStore = isValJson ? JSON.stringify(value) : value;
+      const bytes = Buffer.byteLength(valueToStore, "utf-8");
+      await this.client.setex(key,dataExpiry, valueToStore);
+      stats.gauge('redis_set_val_size', bytes, {
+        timestamp: Date.now()
+      });
+    } catch (e) {
+      throw new RedisError(`Error setting value in Redis due ${e}`);
+    }
+  },
+  async disconnect() {
+    if (process.env.USE_REDIS_DB !== 'false') {
```

Comment:
```suggestion
    if (process.env.USE_REDIS_DB && process.env.USE_REDIS_DB !== 'false') {
```

Reply:
Only for `disconnect`,  `USE_REDIS_DB` flag check is required to know whether redis disconnection should be handled as part of graceful shutdown.

For `init`, ideally flag check shouldn't be required as the calling function (shopify/transform.js) will control whether to use redis or not. Any request coming to `getVal` should be definitely served by initialising connection.
cc: @koladilip

Reply:
When `RedisDB` is imported in the calling function (shopify/transform.js) then class intialization is taking place automatically independant of if we are making any get or set cals to redis. Thats why USE_REDIS_DB flag check is used.

Reply:
Little refactoring is required to avoid this. Since the flag existence is temporary, this should be fine. Incorporate the suggestion & should be good to go.

Reply:
Done